storage/innobase/srv/srv0srv.cc

```diff
@@ -1710,6 +1710,59 @@ void srv_update_purge_thread_count(uint n)
 
 Atomic_counter<int> srv_purge_thread_count_changed;
 
+/** get age of the redo log in lsn units
+@return percentage of redo log filled (calculated using lsn) */
+static lsn_t get_lsn_age_factor()
+{
+  mysql_mutex_lock(&buf_pool.flush_list_mutex);
+  lsn_t age = (log_sys.get_lsn() - buf_pool.get_oldest_modification(0));
+  mysql_mutex_unlock(&buf_pool.flush_list_mutex);
+  return ((age * 100) / log_sys.max_checkpoint_age);
+}
```

Comment:
The type `lsn_t` is supposed to be a semantic data type, associated with LSN only. The `age` in this function should better be of a different type. Yes, in some other code we may use the `lsn_t` data type for differences of LSN.

Which age are we actually determining? If the `buf_pool.flush_list` is empty, we would seem to have `age=log_sys.lsn`. That feels wrong to me. Would we not want something like the following instead:
```c++
  mysql_mutex_lock(&buf_pool.flush_list_mutex);
  lsn_t lsn= log_sys.get_lsn();
  uint64_t age= lsn - buf_pool.get_oldest_modification(lsn);
  mysql_mutex_unlock(&buf_pool.flush_list_mutex);
```
With the above, we would have the intuitive value `age=0` if `buf_pool.flush_list` is empty. That I would understand as ‘checkpoint age’.

Still, it is theoretically possible that `age > log_sys.max_checkpoint_age`. Maybe we should cap the return value to a maximum in that case.

I would suggest that the return value be 0 to 1 less than a power of 2, with a matching type (at most `uint32_t` or `size_t`). There is no need to force the use of a 64-bit type on 32-bit systems.

Reply:
1. replaced the logic to use checkpoint age as that is what we care for
2. replaced data-type to used ulint which should take care of things on 32 bits.
3. we are expecting the % of redo log filed that user can also co-related with the stats they see using show status. re-mapping them to 0-1 or some other base system would break this co-relation and not sure what advantage we get vs 0-100 base system.